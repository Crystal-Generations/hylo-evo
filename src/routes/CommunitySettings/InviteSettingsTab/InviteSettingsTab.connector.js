import { connect } from 'react-redux'
import { communityJoinUrl } from 'util/index'
// import getMe from 'store/selectors/getMe'

export function mapStateToProps (state, props) {
  const { community } = props
  const inviteLink = communityJoinUrl(community)

  return {
    inviteLink
  }
}

export const mapDispatchToProps = {}

export default connect(mapStateToProps, mapDispatchToProps)
