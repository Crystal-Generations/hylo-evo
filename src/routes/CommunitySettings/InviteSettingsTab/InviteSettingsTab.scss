.exampleName { /* !empty */ }

.header {
  border-radius: 4px;
  background-color: $color-athens-gray;
  padding: $space-7x $space-6x;
  margin-bottom: $space-8x;
  margin-top: $space-6x;
}

.title {
  composes: hdr-display from 'css/typography.scss';
}

.subtitle {
  composes: caption-lt-lg from 'css/typography.scss';
}

.invite-link-settings {
  display: flex;
  align-items: flex-end;
}

.invite-link-text {
  display: flex;
  flex-direction: column;
}

.help {
  composes: caption-lt-lg from 'css/typography.scss';
  margin-bottom: 10px;
}

.invite-link {
  composes: text-button from 'css/typography.scss';
  font-size: 13px;
}

.buttons {
  margin-left: auto;
}

.button {
  padding: 0 15px;
}

.reset-button {
  composes: button;
  margin-right: 10px;
}
