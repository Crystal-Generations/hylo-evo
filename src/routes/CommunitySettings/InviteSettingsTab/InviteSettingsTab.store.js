import { createSelector as ormCreateSelector } from 'redux-orm'
import orm from 'store/models'

export const MODULE_NAME = 'InviteSettingsTab'
export const CREATE_INVITATIONS = `${MODULE_NAME}/CREATE_INVITATIONS`
export const CREATE_INVITATIONS_PENDING = `${MODULE_NAME}/CREATE_INVITATIONS_PENDING`

export const EXPIRE_INVITATION = `${MODULE_NAME}/EXPIRE_INVITATION`
export const EXPIRE_INVITATION_PENDING = `${MODULE_NAME}/EXPIRE_INVITATION_PENDING`

export const RESEND_INVITATION = `${MODULE_NAME}/RESEND_INVITATION`
export const RESEND_INVITATION_PENDING = `${MODULE_NAME}/RESEND_INVITATION_PENDING`

export const REINVITE_ALL = `${MODULE_NAME}/REINVITE_ALL`
export const REINVITE_ALL_PENDING = `${MODULE_NAME}/REINVITE_ALL_PENDING`

const defaultState = []

export default function reducer (state = defaultState, action) {
  const { error, type } = action
  if (error) return state

  switch (type) {
    default:
      return state
  }
}

export function createInvitations (communityId, emails, message) {
  return {
    type: CREATE_INVITATIONS,
    graphql: {
      query: `mutation ($communityId: ID, $data: InviteInput) {
        createInvitation(communityId: $communityId, data: $data) {
          invitations {
            id,
            email,
            created_at,
            last_sent_at,
            error
          }
        }
      }`,
      variables: {
        communityId,
        data: {
          emails,
          message
        }
      }
    },
    meta: {
      communityId,
      emails,
      optimistic: true
    }
  }
}

export function reinviteAll (communityId) {
  return {
    type: REINVITE_ALL,
    graphql: {
      query: `mutation ($communityId: ID) {
        reinviteAll(communityId: $communityId) {
          success
        }
      }`,
      variables: {
        communityId
      }
    },
    meta: {
      communityId,
      optimistic: true
    }
  }
}

export function expireInvitation (invitationToken) {
  return {
    type: EXPIRE_INVITATION,
    graphql: {
      query: `mutation ($invitationToken: ID) {
        expireInvitation(invitationToken: $invitationToken) {
          success
        }
      }`,
      variables: {
        invitationToken
      }
    },
    meta: {
      invitationToken,
      optimistic: true
    }
  }
}

export function resendInvitation (invitationToken) {
  return {
    type: RESEND_INVITATION,
    graphql: {
      query: `mutation ($invitationToken: ID) {
        resendInvitation(invitationId: $invitationToken) {
          success
        }
      }`,
      variables: {
        invitationToken
      }
    },
    meta: {
      invitationToken,
      optimistic: true
    }
  }
}

// expects props to be of the form {communityId}
export const getPendingInvites = ormCreateSelector(
  orm,
  state => state.orm,
  (state, props) => props.communityId,
  ({ Community }, id) => {
    const community = Community.safeGet({id})
    if (!community || !community.pendingInvitations) return []
    community.pendingInvitations.orderBy(i => -new Date(i.created_at)).toModelArray()
    return community
    .pendingInvitations
    .orderBy(i => -new Date(i.created_at))
    .filter(invite => invite.id)
    .toModelArray()
  }
)

export function ormSessionReducer (session, { type, meta, payload }) {
  const { Community, Invitation } = session
  let community, invite

  switch (type) {
    case CREATE_INVITATIONS:
      community = Community.withId(meta.communityId)
      let invites = payload.data.createInvitation.invitations.map(i => Invitation.create(i))

      community.updateAppending({pendingInvitations: invites})
      break

    case RESEND_INVITATION_PENDING:
      invite = Invitation.withId(meta.invitationToken)
      if (!invite) break
      invite.update({resent: true, last_sent_at: new Date()})
      break

    case EXPIRE_INVITATION_PENDING:
      invite = Invitation.withId(meta.invitationToken)
      invite.delete()
      break

    case REINVITE_ALL_PENDING:
      community = Community.withId(meta.communityId)
      community.pendingInvitations.update({resent: true, last_sent_at: new Date()})
      break
  }
}
