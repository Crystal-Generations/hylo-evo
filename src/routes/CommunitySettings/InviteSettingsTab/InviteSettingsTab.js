import React from 'react'
import './InviteSettingsTab.scss'
import Button from 'components/Button'
import CopyToClipboard from 'react-copy-to-clipboard'

const { object } = React.PropTypes

export default function InviteSettingsTab ({ community, resetLink, inviteLink }) {
  const { name } = community
  // const inviteLink = 'https://www.hylo.com/c/hylo/join/aelaknetiken'

  const onReset = () => {
    if (window.confirm("Are you sure you want to create a new join link? The current link won't work anymore if you do.")) {
      resetLink()
    }
  }
  const onCopy = () => console.log('copied')

  return <div>
    <div styleName='header'>
      <div styleName='title'>Invite People</div>
      <div styleName='subtitle'>
        to {name} on Hylo
      </div>
    </div>
    <div styleName='invite-link-settings'>
      <div styleName='invite-link-text'>
        <div styleName='help'>Anyone with this link can join the community</div>
        <div styleName='invite-link'>{inviteLink}</div>
      </div>
      <div styleName='buttons'>
        <Button onClick={onReset}
          color='green-white-green-border'
          styleName='reset-button'
          narrow
          small>
          Reset Link
        </Button>
        <CopyToClipboard text={inviteLink} onCopy={onCopy}>
          <Button color='green-white-green-border' styleName='button' narrow small>
            Copy Link
          </Button>
        </CopyToClipboard>
      </div>
    </div>
  </div>
}
InviteSettingsTab.propTypes = {
  community: object
}
