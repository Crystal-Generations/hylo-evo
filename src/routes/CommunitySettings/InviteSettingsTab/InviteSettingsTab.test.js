import InviteSettingsTab from './InviteSettingsTab'
import { shallow } from 'enzyme'
import React from 'react'

it('does something', () => {
  const wrapper = shallow(<InviteSettingsTab />)
  expect(wrapper).toMatchSnapshot()
})
