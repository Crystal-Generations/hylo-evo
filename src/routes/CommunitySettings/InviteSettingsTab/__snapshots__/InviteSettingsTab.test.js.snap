// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="title hdr-display"
    >
      Invite People
    </div>
    <div
      className="subtitle caption-lt-lg"
    >
      to 
      Hylo
       on Hylo
    </div>
  </div>
  <div
    className="invite-link-settings"
  >
    <div
      className="invite-link-text"
    >
      <div
        className="help caption-lt-lg"
      >
        Anyone with this link can join the community
      </div>
      <div
        className="invite-link text-button"
      >
        http://www.hylo.com/c/hylo/join/lalala
      </div>
    </div>
    <div
      className="buttons"
    >
      <Button
        className="reset-button button"
        color="green-white-green-border"
        narrow={true}
        onClick={[Function]}
        small={true}
      >
        Reset Link
      </Button>
      <CopyToClipboard
        onCopy={[Function]}
        text="http://www.hylo.com/c/hylo/join/lalala"
      >
        <Button
          className="copy-button button"
          color="green-white-green-border"
          narrow={true}
          small={true}
        >
          Copy Link
        </Button>
      </CopyToClipboard>
    </div>
  </div>
  <div
    className="email-section"
  >
    <TextareaAutosize
      className="invite-msg-input"
      minRows={5}
      onChange={[Function]}
      onHeightChange={[Function]}
      placeholder="Type email addresses"
      useCacheForDOMMeasurements={false}
      value=""
    />
    <TextareaAutosize
      className="invite-msg-input"
      minRows={5}
      onChange={[Function]}
      onHeightChange={[Function]}
      useCacheForDOMMeasurements={false}
      value="Hi!

I'm inviting you to join Hylo community on Hylo.

Hylo is using Hylo for our online community: this is our dedicated space for communication & collaboration."
    />
    <div
      className="send-invite-button"
    >
      <Button
        color="green"
        disabled={true}
        narrow={true}
        onClick={[Function]}
        small={true}
      >
        Send Invite
      </Button>
    </div>
  </div>
  <div
    className="pending-invites-section"
  >
    <div
      className="pending-invites-header"
    >
      <h1
        style={
          Object {
            "flex": 1,
          }
        }
      >
        No 
        Pending Invites
      </h1>
    </div>
    <div
      className="pending-invites-list"
    >
      <CSSTransitionGroup
        transitionAppear={false}
        transitionEnter={true}
        transitionEnterTimeout={400}
        transitionLeave={true}
        transitionLeaveTimeout={500}
        transitionName={
          Object {
            "enter": undefined,
            "enterActive": undefined,
            "leave": undefined,
            "leaveActive": undefined,
          }
        }
      />
    </div>
  </div>
</div>
`;
